tests/fixtures/webext-as-library/helpers.js

```diff
@@ -1,15 +1,34 @@
 const assert = require('assert');
 
-function testWebExtModuleImports(webExt) {
+function testModuleExports(webExt) {
   assert.deepEqual(Object.keys(webExt).sort(), ['cmd', 'main', 'util'].sort());
   assert.deepEqual(Object.keys(webExt.util).sort(), ['logger', 'adb'].sort());
+  assert.equal(typeof webExt.cmd.run, 'function');
+
+  assertImportedADB({expectLoaded: false});
   assert.deepEqual(
     Object.keys(webExt.util.adb).sort(),
     ['listADBDevices', 'listADBFirefoxAPKs'].sort(),
   );
-  assert.equal(typeof webExt.cmd.run, 'function');
+  assertImportedADB({expectLoaded: true});
+}
+
+function assertImportedADB({expectLoaded}) {
+  const hasAdbDeps = Object.keys(require.cache).filter(
+    (filePath) => filePath.includes('@devicefarmer/adbkit')
+  ).length > 0;
+
+  const msg = expectLoaded
+    ? 'adb module should have been loaded'
+    : 'adb module should not be loaded yet';
+  // eslint-disable-next-line no-console
+  console.log(Object.keys(require.cache).filter(
+    (filePath) => filePath.includes('@devicefarmer/adbkit')
+  ));
```

Comment:
@ankushduacodes I guess that the console.log here is meant to help investigating why the test is failing when running on windows, right?

The string logged here to stdout is not visible yet in the CI job failure because it is called in the child process that `test.lib.imports.js` is spawning using [`child_process.execFileSync`](https://nodejs.org/api/child_process.html#child_process_child_process_execfilesync_file_args_options) nodejs API method and only the errors are going to be printed by default by the parent process, quoting from the docs linked above:
> stderr by default will be output to the parent process' stderr...

A quick way to make sure that the child process stdout is also printed automatically on the parent process stdout is to change those calls temporarily to include `stdio: 'inherit'` ([see options.stdio doc here to know more about the values allowed for this option](https://nodejs.org/api/child_process.html#child_process_options_stdio)), that should be enough to let us see this debugging logs in the circleci job logs:

```
      execFileSync(node, ['--experimental-modules', 'test-import.mjs'], {
        cwd: tmpDir.path(),
        stdio: 'inherit',
      });
```

would you mind to try this so that we may have some more details to figure out why the test is failing on windows?
(if even this doesn't provide us enough details then I'll reboot one of my machine and try it out on a local windows system)

Reply:
@rpl I was setting up sideloading  windows on my machine to test this out. And I have set it up successfully, I will update you soon on whats going on.

Reply:
@rpl The issue here seems to be with the difference in the way paths are written in UNIX/LINUX ie. using forward slash `/` but on windows we use backward slash `\` which then needs to escaped using another backward slash.
Please check out the following logs:
```
PS C:\Users\ankus\Documents\GitHub\web-ext> node
Welcome to Node.js v14.15.4.
Type ".help" for more information.
> const webExt = require('./dist/web-ext')
undefined
> Object.keys(require.cache)
[
  'C:\\Users\\ankus\\Documents\\GitHub\\web-ext\\dist\\web-ext.js',
  'C:\\Users\\ankus\\Documents\\GitHub\\web-ext\\node_modules\\source-map-support\\source-map-support.js',
  ...
]
> webExt.util.adb
{
  listADBDevices: [AsyncFunction: listADBDevices],
  listADBFirefoxAPKs: [AsyncFunction: listADBFirefoxAPKs]
}
> Object.keys(require.cache).filter(filepath => filepath.includes('@devicefarmer\\adbkit'))
[
  'C:\\Users\\ankus\\Documents\\GitHub\\web-ext\\node_modules\\@devicefarmer\\adbkit\\index.js',
  'C:\\Users\\ankus\\Documents\\GitHub\\web-ext\\node_modules\\@devicefarmer\\adbkit\\lib\\adb.js',
  ...
]
```

Reply:
@ankushduacodes thanks for investigating that windows failure, what you described does definitely make sense and expected.

Instead of conditioning the string we look for in the cache based on the current platform, could we use path.join to get the platform specific format on both the platforms? something like this:

```
const path = require('path');
const adbPathString = path.join('@devidefarmer', 'adbkit');
...
```

would you mind to give that a try?

> (PS: Sorry for delay in response past few days, I got caught up in school work as my finals will be starting soon, I hope you understand :)).

Don't worry at all, school work comes first!
And anyway in my opinion you are doing great (you are always following up to what we discuss and agree on, and you have been pretty responsive from my perspective). I'm really enjoying working with you.

Reply:
> Instead of conditioning the string we look for in the cache based on the current platform, could we use path.join to get the platform specific format on both the platforms? something like this:
> 
> ```
> const path = require('path');
> const adbPathString = path.join('@devidefarmer', 'adbkit');
> ...
> ```
> 
> would you mind to give that a try?

For sure!..

> Don't worry at all, school work comes first!

Thank you for your understanding.

> And anyway in my opinion you are doing great (you are always following up to what we discuss and agree on, and you have been pretty responsive from my perspective). I'm really enjoying working with you.

And Thank you for the kind words. I really appreciate that ❤️. I really enjoy working with you as well.